File: SmartWash/src/main/java/com/smartwash/vo/coupon/CouponVo.java
package com.smartwash.vo.coupon;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
public class CouponVo {
    private Long couponId;

    private String title;

    private String description;

    private BigDecimal discount;

    private BigDecimal threshold;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime startTime;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime endTime;

    private Integer validDays;

    private Boolean isNewUser;

    private String status;
//    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
//    private LocalDateTime createdAt;
//    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
//    private LocalDateTime updatedAt;
}

File: SmartWash-Android/app/src/main/java/com/smartwash/network/vo/coupon/CouponVo.kt
package com.smartwash.network.vo.coupon

import androidx.annotation.Keep

@Keep
data class CouponVo(
    val couponId: Long,
    val title: String,
    val description: String,
    val discount: Float,
    val threshold: Float,
    val startTime: String,
    val endTime: String,
    val validDays: Int,
    val isNewUser: Boolean,
    val status: String,
)
